### ISSUE TITLE
Uncaught (in promise) ReferenceError: WebAssembly is not defined

### ISSUE THREAD

**Issue by emanruse** [uBlockOrigin/uBlock-issues#900]
### Prerequisites

- [x] I verified that this is not a filter issue
- [x] This is not a support issue or a question
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

Chromium shows "Errors" when installing through "Load unpacked" the unpacked uBlock0_1.25.0.chromium.zip. Clicking that "Errors" button shows a message:

```
Uncaught (in promise) ReferenceError: WebAssembly is not defined
```

and a stack trace:

```
js/strie.js:682 (anonymous function)
```

### A specific URL where the issue occurs

chrome://extensions/

### Steps to Reproduce

1. Run Chromium with command line option `--js-flags=--noexpose_wasm` (because by default I don't want websites to run WASM on my computer)
2. Download and unzip uBlock0_1.25.0.chromium.zip
3. "Load unpacked" the result of step 2.

#### Disclaimer regarding expected/actual behavior:

I don't know if the displayed error results in something not working or if it has just performance effect (which I can't assess) due to JS being slower than WASM.

### Expected behavior:

My general expectation is to be able to:

- run the WASM code which I trust explicitly (e.g. an extension installed through "Load unpacked")
- block every other WASM code

just like one can block JS and unblock it on specific sites only.

Unfortunately I am unaware if such separation is possible, if it is up to the extension or anything else. In any case I don't see "per-site" control of WASM in the browser itself or in uBO. Might that be an enhancement of uBO or uMatrix to discuss separately? If yes - please let me know and I will open a new issue.

### Actual behavior:

As per description.

Visually uBO seems to work and does block content. Without the command line option `--js-flags=--noexpose_wasm` the error message doesn't show.

### Your environment

* uBlock Origin version: 1.25.0
* Browser Name and version: ungoogled-chromium 80.0.3987.106-1.1 (portable)
* Operating System and version: openSUSE Leap 15.1

**Comment by uBlock-user** [comment]
> Run Chromium with command line option --js-flags=--noexpose_wasm (because by default I don't want websites to run WASM on my computer)

You're disabling WASM so ofcourse it will throw this error.

**Comment by uBlock-user** [comment]
> run the WASM code which I trust explicitly (e.g. an extension installed through "Load unpacked")
    block every other WASM code

Not possible in any browser.

**Comment by emanruse** [comment]
@uBlock-user, thank you for the quick feedback.

Could you please answer if this error means that something in uBO is not working -or- it simply means "Everything works, just slower because only JS is used".

> Not possible in any browser.

Is that (per site WASM enable/disable) a feature which can be implemented in an extension (uBO or uM)? If it is technically possible I would take the time to suggest it in a separate issue.

**Comment by uBlock-user** [comment]
> Could you please answer if this error means that something in uBO is not working -or- it simply means "Everything works, just slower because only JS is used".

@emanruse uBO makes use of WASM and the browser doesn't send any WASM calls to the JS engine because you disabled WASM explicitly with that flag, which ultimately results in browser throwing a runtime error like this one.

**Comment by emanruse** [comment]
@uBlock-user, I understand that. But the question is: does this make uBO (partly) dysfunctional or simply slower?

**Comment by uBlock-user** [comment]
> does this make uBO (partly) dysfunctional

Yes, upto some extent.

**Comment by emanruse** [comment]
> Yes, upto some extent.

Where can I get info about the actual extent? Need I ask on Reddit?

**Comment by gorhill** [comment]
I do test for presence of WebAssembly and fall back to JS when not available. However in the current case I test after trying to use it, this is a mistake.

**Comment by emanruse** [comment]
> I do test for presence of WebAssembly and fall back to JS when not available. 

Does that mean that the answer given by @uBlock-user ("Yes, upto some extent.") is incorrect and that no functionality is actually hurt but the extension fulfills its functions, only through JS and not through WASM? I.e. that it is fully functional just slower.

> this is a mistake.

Does that mean there is an actual problem to fix and that this issue should be reopened?

**Comment by mapx-** [comment]
yes + yes

**Comment by uBlock-user** [comment]
>  @uBlock-user ("Yes, upto some extent.")

Any WASM based functionality won't work obviously though the falling back on JS will not cripple uBO.

**Comment by emanruse** [comment]
Thank you for clarifying.

**Comment by Thorin-Oakenpants** [comment]
FYI: for Firefox 71+ users, disabling wasm (`javascript.options.wasm`=`false`) will no longer impact extensions [1576254](https://bugzilla.mozilla.org/show_bug.cgi?id=1576254)

> I do test for presence of WebAssembly

not sure if `1576254` affects your tests (for FF)
